etc/docs/build.ts

```diff
@@ -0,0 +1,96 @@
+#! /usr/bin/env ts-node
+
+import { parse, stringify } from '@iarna/toml';
+import { exec as execCb } from 'child_process';
+import { writeFile, readFile } from 'fs/promises';
+import { promisify } from 'util';
+import { chdir } from 'process';
+import { confirm, customSemverCompare, getCommandLineArguments, JsonVersionSchema, TomlVersionSchema, VersionSchema } from './utils';
+
+const exec = promisify(execCb);
+
+const RELEASES_TOML_FILE = './template/data/releases.toml';
+const RELEASES_JSON_FILE = './template/static/versions.json';
+
+const copyGeneratedDocsToDocsFolder = () => exec(`cp -R temp/. ../../docs/.`);
+const removeTempDirectory = () => exec('rm -rf temp');
+const installDependencies = () => exec('npm i --no-save typedoc');
+const buildDocs = () => exec('npm run build:typedoc');
+
+async function copyNewDocsToGeneratedSite({ tag }: VersionSchema) {
+  const outputDirectory = `./temp/${tag}`;
+  const pathToBuiltDocs = './build';
+  const command = `cp -R ${pathToBuiltDocs} ${outputDirectory}`;
+  return await exec(command);
+}
+
+async function updateSiteTemplateForNewVersion(newVersion: VersionSchema, tomlData: TomlVersionSchema, jsonVersions: JsonVersionSchema[]) {
+  const versionExists = jsonVersions.some(({ version }) => version === newVersion.tag);
+  if (versionExists) {
+    const existingVersionIndex = tomlData.versions.findIndex(({ tag }) => tag === newVersion.tag);
+    tomlData.versions[existingVersionIndex] = newVersion;
+  } else {
+    tomlData.versions.unshift(newVersion);
+
+    jsonVersions.unshift({ version: newVersion.tag })
+  }
+
+  tomlData.versions.sort((a, b) => customSemverCompare(a.tag, b.tag))
+  tomlData.current = tomlData.versions[0].version;
```

Comment:
Just noticed this part - will it be an issue when we update the docs with the `next` version? Maybe we should just create the `next` docs in this PR, and then update this to be `tomlData.versions[1].version` (or we can keep it [0] but filter out "next")